core/src/main/java/io/grpc/ServerCall.java

```diff
@@ -176,4 +176,16 @@ public boolean isReady() {
   public void setMessageCompression(boolean enabled) {
     // noop
   }
+
+  /**
+   * Sets the compression algorithm for this call.  If the server does not support the compression
+   * algorithm, the call will fail.
+   *
+   * @param compressor the name of the compressor to use.
+   * @throws IllegalArgumentException if the compressor name can not be found.
+   */
+  @ExperimentalApi
+  public void setCompression(String compressor) {
+    throw new UnsupportedOperationException();
```

Comment:
Add docs, and updated ForwardingServerCall.  UnsupportedOperationException will only be thrown by ServerCalls that forgot to override it.